test/1_stdlib/UnfoldSequence.swift

```diff
@@ -0,0 +1,41 @@
+// RUN: %target-run-simple-swift
+// REQUIRES: executable_test
+
+import StdlibUnittest
+
+let UnfoldTests = TestSuite("UnfoldSequence")
+
+UnfoldTests.test("sequence(state:next:)") {
+  // FIXME: The full type signatures on these closures should not be
+  // necessary, but at the moment the compiler gives very confusing errors if
+  // we don't have them.
+  let s0 = sequence(state: 0, next: { (val: inout Int) -> Int? in val += 1; return val < 10 ? val : nil })
+  expectEqualSequence(1..<10, s0)
+
+  let s1 = sequence(state: (), next: { (_: inout ()) -> Int? in 1 })
+  expectEqualSequence([1, 1, 1], s1.prefix(3))
```

Comment:
I think we need to start writing more tests for sequence methods that should operate on infinite sequences.  `prefix()` is one of them, but there are a lot more.